src/main/java/com/ericsson/gerrit/plugins/eiffel/events/generators/LinkGenerator.java

```diff
@@ -0,0 +1,137 @@
+
+        addIfNotEmpty(LINK_TYPE_PREVIOUS_VERSION, previousEvent);
+    }
+
+    /**
+     * Create a PREVIOUS_VERSION link to the parent of the given commit id
+     *
+     * @param commitId of the parent we should link to
+     */
+    public void addScsPreviousVersion(final String commitId) {
+        final List<String> parentsSHAs = commitInformation.getParentsSHAs(commitId,
+                pluginConfig.getProject());
+        final String previousEvent = getPreviousEiffelEventId(EventType.SCS_EVENT, parentsSHAs);
+
+        addIfNotEmpty(LINK_TYPE_PREVIOUS_VERSION, previousEvent);
+    }
+
+    /**
+     * Create a BASE link to the parent of the given commit id
+     *
+     * @param commitId of the parent we should link to
+     */
+    public void addSccBase(final String commitId) {
+        final List<String> parentsSHAs = commitInformation.getParentsSHAs(commitId,
+                pluginConfig.getProject());
+        final String previousEvent = getPreviousEiffelEventId(EventType.SCS_EVENT, parentsSHAs);
+
+        addIfNotEmpty(LINK_TYPE_BASE, previousEvent);
+    }
+
+    /**
+     * Create a CHANGE link to the last event for the given changeId
+     *
+     * @param changeId for event to link to
+     */
+    public void addScsChange(final String changeId) {
+        final String previousEvent = getPreviousEiffelEventId(EventType.SCC_EVENT, changeId);
+
+        addIfNotEmpty(LINK_TYPE_CHANGE, previousEvent);
+    }
+
+    /**
+     * Get the generated links
+     *
+     * @return the links
+     */
+    public ArrayList<Link> generateLinks() {
+        return links;
+    }
+
+    private void addIfNotEmpty(final String eventType, final String previousEvent) {
+        if (!StringUtils.isEmpty(previousEvent)) {
+            final Link link = new Link();
+            link.type = eventType;
+            link.target = previousEvent;
+            links.add(link);
+        }
+    }
+
+    /**
+     * Will for a given list of search criteria return the first found event
+     */
+    private String getPreviousEiffelEventId(final String linkedEiffelEventType,
+            final List<String> searchCriterias) {
+        for (final String searchCriteria : searchCriterias) {
+            final String eiffelEventId = getPreviousEiffelEventId(linkedEiffelEventType,
+                    searchCriteria);
+            if (!StringUtils.isEmpty(eiffelEventId)) {
+                return eiffelEventId;
+            }
+        }
+        return "";
+    }
+
+    private String getPreviousEiffelEventId(final String linkedEiffelEventType,
+            final String searchCriteria) {
+        try {
+            final EventStorage eventStorage = EventStorageFactory.getEventStorage(
+                    pluginConfig, linkedEiffelEventType);
+            return eventStorage.getEventId(pluginConfig.getProject(), searchCriteria);
+        } catch (final IllegalArgumentException e) {
+            LOGGER.error("Could not get previous Eiffel event.", e);
```

Comment:
This could be a single catch and the message logged could be from the thrown exception.

Reply:
The reason for multiple catches is to give more information. Why would you like just one?

Reply:
If the thrown exceptions message is self explaining more information would not be needed. It reduces the number of catches. The code looks better.

Reply:
That would require changes and digging in other places in the code and I don't want to do that in this pullrequest.